full-service/src/db/transaction_log.rs

```diff
@@ -1968,4 +1970,63 @@ mod tests {
 
         assert_eq!(tx_log.tx, expected_tx_log.tx);
     }
+
+    #[test]
+    fn test_try_from_vec_output_txo_for_transaction_id() {
+        let mut rng: StdRng = SeedableRng::from_entropy();
+        let root_id = RootIdentity::from_random(&mut rng);
+        let recipient_account_key = AccountKey::from(&root_id);
+
+        let num_txos = 5; // txo vector size
+        let num_loops = 5; // number of times to run test
+        let amount = 77;
+
+        for _ in 1..=num_loops {
+            // run test loop
+
+            let mut output_vec: Vec<OutputTxo> = Vec::new();
+
+            for _ in 0..num_txos {
```

Comment:
🔧 I'm not fully understanding the purpose for `1..=num_loops` I don't think it gains much.
Ideally one should have test cases for  zero, one, multiple, the "zom" of [zombies](https://medium.com/@simontoth/daily-bit-e-of-c-testing-zombies-504238a18ef9).

consider:
```suggestion
        for loop_number in 1..=num_loops {

            let mut output_vec: Vec<OutputTxo> = Vec::new();

            for _ in 1..=loop_number {
```
this would get the one and many scenarios, with a few "many" scenarios
but I "personally" would probably lean toward parameterized tests 
The advantage of parameterized tests is each one runs independently so you could get more info about whats failing. 
With a loop the first failure stops the test, and thus all subsequent loop iterations
```rust
    #[parameterized(
        one = {1},
        two = {2},
        three = {3},
        ten = {10}
    )]
    fn test_try_from_vec_output_txo_for_transaction_id(num_txos: usize) {
        let mut rng: StdRng = SeedableRng::from_entropy();
        let root_id = RootIdentity::from_random(&mut rng);
        let recipient_account_key = AccountKey::from(&root_id);
        let amount = 77;
        let mut output_vec: Vec<OutputTxo> = Vec::new();

        for _ in 0..num_txos {
```

Reply:
Because the test txos are being initialized from entropy, I wanted to run the test more than once.   The `1..=num_loops` was just a simple way to do that.  But I see what you are showing is much more powerful.  I took a quick look at **yare** just now, it seems pretty straightforward.  And the 🧟‍♂️ mnemonic is good too!

Reply:
I tried to go the `yare` route on this one.  I was able to get the crate loaded, apply the code above, and build cleanly.  But I couldn't get the test to run after that.  I had the name of the test on the command line, and as I'm writing this I'm wondering if maybe I need to either change the test name in some way, or just give no test name and see if it runs along with all the tests.

In any case, I'm falling back to the nested looping in the form Nick suggested prior to mentioning `yare` above.  That seems to work quite well, even without being parameterized.